### ISSUE TITLE
compose v2 ignores the --force-recreate option in `docker-compose up` command.

### ISSUE THREAD

**Issue by pratiksanglikar** [docker/compose#8606]
**Description**
When using docker-compose CLI v2, the `docker-compose up` command ignores the `--force-recreate` option and still uses the old container configuration if the container is already present.

**Steps to reproduce the issue:**
1. Create a sample docker-compose project using the [getting started tutorial](https://docs.docker.com/compose/gettingstarted/). *Else clone [pratiksanglikar/compose-sample](https://github.com/pratiksanglikar/compose-sample) and jump to step 3.*
2. Modify the `docker-compose.yml` file to add `container_name` to the service `web`.
The final `docker-compose.yml` file looks like - 
```yml
version: "3.9"
services:
  web:
    container_name: python_webapp
    build: .
    ports:
      - "5000"
      - "6000"
  redis:
    image: "redis:alpine"
``` 
3. Start the app using `docker-compose up` command.
4. Inspect the container name created for the python app.
```
compose-sample> docker ps
CONTAINER ID   IMAGE                COMMAND                  CREATED          STATUS         PORTS                                              NAMES
9543330c17b7   redis:alpine         "docker-entrypoint.s…"   10 seconds ago   Up 8 seconds   6379/tcp                                           compose-sample_redis_1
837dd056c771   compose-sample_web   "flask run"              10 seconds ago   Up 8 seconds   0.0.0.0:62279->5000/tcp, 0.0.0.0:62280->6000/tcp   python_webapp
```
5. Force kill the compose stack by pressing `Ctrl+C` twice.
6. Change the `container_name` to `python_webapp_1` in the `docker-compose.yml` file.
7. The final `docker-compose.yml` file looks like this -
```yml
version: "3.9"
services:
  web:
    container_name: python_webapp_1
    build: .
    ports:
      - "5000"
      - "6000"
  redis:
    image: "redis:alpine"
```
8. Start the compose stack again with `--force-recreate` option. i.e. execute `docker-compose up --force-recreate`.
9. Check the container name for the python app using command `docker ps`.

**Describe the results you received:**
```
compose-sample> docker ps
CONTAINER ID   IMAGE                COMMAND                  CREATED          STATUS         PORTS                                              NAMES
aece53cca2a7   redis:alpine         "docker-entrypoint.s…"   10 seconds ago   Up 8 seconds   6379/tcp                                           compose-sample_redis_1
12a3b1243b03   compose-sample_web   "flask run"              10 seconds ago   Up 8 seconds   0.0.0.0:62279->5000/tcp, 0.0.0.0:62280->6000/tcp   python_webapp
```
Notice the name of the container with id `12a3b1243b03` is `python_webapp`.

**Describe the results you expected:**
```
compose-sample> docker ps
CONTAINER ID   IMAGE                COMMAND                  CREATED          STATUS         PORTS                                              NAMES
aece53cca2a7   redis:alpine         "docker-entrypoint.s…"   10 seconds ago   Up 8 seconds   6379/tcp                                           compose-sample_redis_1
12a3b1243b03   compose-sample_web   "flask run"              10 seconds ago   Up 8 seconds   0.0.0.0:62279->5000/tcp, 0.0.0.0:62280->6000/tcp   python_webapp_1
```
With the `--force-recreate` option provided to the `docker-compose up` command, the container should be recreated taking into account the changes made in the `docker-compose.yml`.
i.e., the name of the container `12a3b1243b03` should be `python_webapp_1` to reflect `container_name` specified in the `docker-compose.yml`.

**Additional information you deem important (e.g. issue happens only occasionally):**
N/A

**Output of `docker-compose --version`:**

```
PS> docker-compose --version
Docker Compose version v2.0.0-rc.2
```

**Output of `docker version`:**

```
PS> docker version
Client:
 Cloud integration: 1.0.17
 Version:           20.10.8
 API version:       1.41
 Go version:        go1.16.6
 Git commit:        3967b7d
 Built:             Fri Jul 30 19:58:50 2021
 OS/Arch:           windows/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.8
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.16.6
  Git commit:       75249d8
  Built:            Fri Jul 30 19:52:10 2021
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.4.9
  GitCommit:        e25210fe30a0a703442421b0f60afac609f950a3
 runc:
  Version:          1.0.1
  GitCommit:        v1.0.1-0-g4144b63
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

**Output of `docker context show`:**  
```
PS> docker context show
default
```

**Output of `docker info`:**

```
PS> docker info
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Build with BuildKit (Docker Inc., v0.6.1-docker)
  compose: Docker Compose (Docker Inc., v2.0.0-rc.2)
  scan: Docker Scan (Docker Inc., v0.8.0)

Server:
 Containers: 36
  Running: 34
  Paused: 0
  Stopped: 2
 Images: 11
 Server Version: 20.10.8
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 io.containerd.runtime.v1.linux runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: e25210fe30a0a703442421b0f60afac609f950a3
 runc version: v1.0.1-0-g4144b63
 init version: de40ad0
 Security Options:
  seccomp
   Profile: default
 Kernel Version: 5.10.43.3-microsoft-standard-WSL2
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64
 CPUs: 12
 Total Memory: 15.56GiB
 Name: docker-desktop
 ID: 4DH2:53OI:XAAU:EI2S:AQOR:HWNR:VAZ6:BVPJ:O773:VYCP:FRM7:JHIJ
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: No blkio throttle.read_bps_device support
WARNING: No blkio throttle.write_bps_device support
WARNING: No blkio throttle.read_iops_device support
WARNING: No blkio throttle.write_iops_device support
```

**Additional environment details (AWS ECS, Azure ACI, local, etc.):**
Host machine: Windows
Containers: Linux

**Comment by ndeloof** [comment]
I was able to reproduce this issue, definitively a bug. Thanks for reporting

**Comment by mat007** [comment]
Fixed by #8609
